packages/cozy-client/src/CozyClient.js

```diff
@@ -402,7 +423,7 @@ export default class CozyClient {
 
   getOrCreateStackClient() {
     if (!this.client) {
-      this.client = new CozyStackClient(this.options)
+      this.client = this.options.oauth ? new OAuthClient(this.options) : new CozyStackClient(this.options)
```

Comment:
I see that `OAuthClient` and `CozyStackClient` return the same concept and take the same parameters. Could it be abstracted in this way?

Reply:
Not sure what you mean by that ? `OAuthClient` is `CozyStackClient` + OAuth. But keeping them separated reduces their individual complexity, and it's transparent for the user.

Reply:
I mean, could we have an unique class that implements what it has to

```
const options = {
  oauth: {}
}
const client = new CozyClient(options)
```

Maybe I misunderstood what you wanted to do here.

Reply:
Yeah, I think it'd be better to let `CozyStackClient` instantiate its OAuth utility if there is an `oauth` option. But it's a simple change that can be done later.

Reply:
I'm afraid it will lead to a lot of if/else-ing inside `CozyStackClient`, but we'll see.